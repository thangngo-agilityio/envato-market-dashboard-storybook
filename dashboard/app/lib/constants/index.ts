export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
export * from './faq';
export * from './avatar';
export * from './sizes';
export * from './setting';
export * from './termAndCondition';
export * from './notification';
export * from './authentication';
export * from './discount';
export * from './metadata';
export * from './og-properties';
export * from './chat';
export * from './product';
