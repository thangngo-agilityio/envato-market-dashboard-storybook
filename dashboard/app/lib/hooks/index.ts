export * from './useForm';
export * from './useAuth';
export * from './useStatistical';
export * from './useTransactions';
export * from './useDebounce';
export * from './useSearch';
export * from './usePagination';
export * from './useStore';
export * from './useUser';
export * from './usePinCode';
export * from './useNotification';
export * from './useWallet';
export * from './useMoney';
export * from './useChat';
export * from './useImageUploader';
export * from './useProducts';
export * from './useRecentActivities';
