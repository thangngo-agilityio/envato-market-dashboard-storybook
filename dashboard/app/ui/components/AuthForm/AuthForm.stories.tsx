import { StoryObj, Meta } from '@storybook/react';

import AuthForm from '.';

const meta: Meta<typeof AuthForm> = {
  title: 'Custom Components/AuthForm',
  tags: ['autodocs'],
  component: AuthForm,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof AuthForm>;

export const Default: Story = {
  args: {},
};

export const RegisterForm: Story = {
  args: {
    isRegister: true,
  },
};
