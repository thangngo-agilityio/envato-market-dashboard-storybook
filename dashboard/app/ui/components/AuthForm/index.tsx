'use client';

import { ChangeEvent, memo, useCallback } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/navigation';
import { ViewOffIcon, ViewIcon } from '@chakra-ui/icons';
import {
  Button,
  HStack,
  Text,
  VStack,
  Checkbox,
  useDisclosure,
  Box,
  Link as ChakraLink,
  Flex,
} from '@chakra-ui/react';

import { Controller, SubmitHandler } from 'react-hook-form';

// Hooks
import { useForm, useAuth } from '@/lib/hooks';

// Constants
import { ROUTES, AUTH_SCHEMA } from '@/lib/constants';

// Components
import { InputField } from '@/ui/components';

// Utils
import { isWindowDefined, validatePassword } from '@/lib/utils';

// Types
import { TUserDetail } from '@/lib/interfaces';
import { getMessaging } from 'firebase/messaging';

// Layouts
import { AuthFooter } from '@/ui/layouts';

// firebase
import { app, requestForToken } from '@/lib/utils';

type TAuthForm = Omit<TUserDetail, 'id' | 'createdAt'> & {
  confirmPassword: string;
  isAcceptPrivacyPolicy: boolean;
  isRemember: false;
};

interface AuthFormProps {
  isRegister?: boolean;
}

const AuthFormComponent = ({
  isRegister = false,
}: AuthFormProps): JSX.Element => {
  const { signIn, signUp } = useAuth();
  const router = useRouter();

  // Control form
  const {
    control,
    formState: {
      errors: { root },
      dirtyFields,
      defaultValues,
      isSubmitting,
    },
    handleSubmit,
    setError,
    clearErrors,
  } = useForm<TAuthForm>({
    defaultValues: {
      email: '',
      password: '',

      ...(!isRegister && {
        isRemember: false,
      }),
      ...(isRegister && {
        firstName: '',
        lastName: '',
        confirmPassword: '',
        isAcceptPrivacyPolicy: false,
      }),
    },
  });

  // Control show/hide password
  const { isOpen: isShowPassword, onToggle: onShowPassword } = useDisclosure();
  const { isOpen: isShowConfirmPassword, onToggle: onShowConfirmPassword } =
    useDisclosure();

  const messaging = isWindowDefined() ? getMessaging(app) : null;

  const isDisabledSubmitBtn: boolean = (() => {
    const getLength = (object: object): number => Object.keys(object).length;

    const isFillAllFields: boolean = isRegister
      ? getLength(dirtyFields) === getLength(defaultValues ?? {})
      : !!(dirtyFields.email && dirtyFields.password);

    return isSubmitting || !isFillAllFields;
  })();

  const renderPasswordIcon = useCallback(
    (isCorrect: boolean, callback: typeof onShowPassword): JSX.Element => {
      const Icon = isCorrect ? ViewIcon : ViewOffIcon;

      return (
        <Icon
          color="gray.400"
          w="25px"
          h="25px"
          cursor="pointer"
          onClick={callback}
        />
      );
    },
    [],
  );

  const handleLogin: SubmitHandler<TAuthForm> = useCallback(
    async (data) => {
      try {
        const { email, password, isRemember } = data;
        const fcmToken =
          (messaging && (await requestForToken(messaging))) || '';

        await signIn({ email, password, fcmToken }, isRemember);

        router.push(ROUTES.ROOT);
      } catch (error) {
        const { message } = error as Error;

        setError('root', { type: 'custom', message });
      }
    },
    [messaging, router, setError, signIn],
  );

  const handleRegister: SubmitHandler<TAuthForm> = useCallback(
    async (data) => {
      const {
        isAcceptPrivacyPolicy: _isAcceptPrivacyPolicy,
        confirmPassword: _confirmPassword,
        ...fieldValues
      } = data;
      try {
        const fcmToken =
          (messaging && (await requestForToken(messaging))) || '';

        await signUp({ ...fieldValues, fcmToken });

        router.push(ROUTES.ROOT);
      } catch (error) {
        const { message } = error as Error;

        setError('root', { type: 'custom', message });
      }
    },
    [messaging, router, setError, signUp],
  );

  const handleClearErrorMessage = useCallback(
    (
      field: keyof TAuthForm,
      isError: boolean,
      onChange: (value: string) => void,
    ) =>
      (data: string) => {
        if (isError) {
          clearErrors(field);
        }

        onChange(data);
      },
    [clearErrors],
  );

  const handleClearRootError = useCallback(
    () => clearErrors('root'),
    [clearErrors],
  );

  return (
    <Box
      id={!isRegister ? 'login-form' : 'register-form'}
      as="form"
      onSubmit={handleSubmit(!isRegister ? handleLogin : handleRegister)}
      w={{
        base: '100%',
        sm: 425,
        md: 460,
      }}
      margin="auto"
      pb={16}
      sx={{
        boxSizing: {
          base: 'border-box',
          md: 'unset',
        },
      }}
    >
      <VStack gap={6}>
        {isRegister && (
          <HStack
            gap={{
              base: 6,
              md: 10,
            }}
            w="100%"
            flexDirection={{
              base: 'column',
              md: 'row',
            }}
          >
            <Controller
              control={control}
              rules={AUTH_SCHEMA.FIRST_NAME}
              name="firstName"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  variant="authentication"
                  placeholder="First name"
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  isDisabled={isSubmitting}
                  onChange={handleClearErrorMessage(
                    'firstName',
                    !!error,
                    field.onChange,
                  )}
                  aria-label="first-name"
                />
              )}
            />
            <Controller
              control={control}
              rules={AUTH_SCHEMA.LAST_NAME}
              name="lastName"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  variant="authentication"
                  placeholder="Last name"
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  isDisabled={isSubmitting}
                  onChange={handleClearErrorMessage(
                    'lastName',
                    !!error,
                    field.onChange,
                  )}
                  aria-label="last-name"
                />
              )}
            />
          </HStack>
        )}
        <Controller
          rules={AUTH_SCHEMA.EMAIL}
          control={control}
          name="email"
          render={({ field: { value, onChange }, fieldState: { error } }) => {
            const handleChange = (valueInput: string) => {
              const sanitizedValue = valueInput.trim();

              onChange(sanitizedValue);
            };

            return (
              <InputField
                variant="authentication"
                placeholder="Email"
                isError={!!error?.message}
                errorMessages={error?.message}
                isDisabled={isSubmitting}
                value={value}
                onChange={handleChange}
                onBlur={handleClearRootError}
              />
            );
          }}
        />

        <Controller
          rules={{ validate: validatePassword }}
          control={control}
          name="password"
          render={({ field, fieldState: { error } }) => (
            <InputField
              type={isShowPassword ? 'text' : 'password'}
              variant="authentication"
              placeholder="Password"
              rightIcon={renderPasswordIcon(isShowPassword, onShowPassword)}
              isError={!!error?.message}
              errorMessages={error?.message}
              isDisabled={isSubmitting}
              {...field}
              onBlur={handleClearRootError}
            />
          )}
        />

        {/* Helpers */}
        {!isRegister && (
          <HStack justifyContent="space-between" w="100%" mt={6}>
            <Controller
              control={control}
              name="isRemember"
              render={({ field: { value, onChange } }) => (
                <Checkbox
                  aria-label="remember"
                  variant="round"
                  isChecked={value}
                  onChange={(e: ChangeEvent<HTMLInputElement>) =>
                    onChange(e.target.checked)
                  }
                  isDisabled={isSubmitting}
                  position="relative"
                >
                  <Text fontWeight="semibold">Remember me</Text>
                </Checkbox>
              )}
            />
            <Button
              as={Link}
              isDisabled={isSubmitting}
              href={`/${ROUTES.FORGOT_PASSWORD}`}
              _hover={{
                bg: 'transparent',
              }}
              bg="transparent"
              w="fit-content"
              aria-label="forgot password"
              color="text.currencyColor"
              fontWeight="semibold"
              textTransform="capitalize"
              textDecoration="underline"
              onClick={(e) => {
                if (isSubmitting) return e.preventDefault();
              }}
            >
              forgot password?
            </Button>
          </HStack>
        )}

        {isRegister && (
          <>
            <Controller
              control={control}
              rules={AUTH_SCHEMA.CONFIRM_PASSWORD}
              name="confirmPassword"
              render={({ field, fieldState: { error } }) => (
                <InputField
                  type={isShowConfirmPassword ? 'text' : 'password'}
                  variant="authentication"
                  placeholder="Confirm password"
                  rightIcon={renderPasswordIcon(
                    isShowConfirmPassword,
                    onShowConfirmPassword,
                  )}
                  {...field}
                  isError={!!error}
                  errorMessages={error?.message}
                  isDisabled={isSubmitting}
                  onChange={handleClearErrorMessage(
                    'confirmPassword',
                    !!error,
                    field.onChange,
                  )}
                />
              )}
            />

            <Flex gap={3}>
              <Controller
                control={control}
                rules={AUTH_SCHEMA.AGREE_POLICY}
                name="isAcceptPrivacyPolicy"
                render={({
                  field: { value, onChange },
                  fieldState: { error },
                }) => (
                  <Checkbox
                    size="md"
                    isChecked={value}
                    isDisabled={isSubmitting}
                    onChange={(e: ChangeEvent<HTMLInputElement>) =>
                      onChange(e.target.checked)
                    }
                    {...(error && {
                      sx: {
                        '> span': {
                          borderColor: 'danger.400',
                        },
                      },
                    })}
                  >
                    <Text
                      color="text.secondary"
                      fontSize={{ base: 'xs', md: 'md' }}
                      flex={1}
                      px={4}
                    >
                      By creating an account, you&apos;re agreeing to our {''}
                      <ChakraLink
                        href="#"
                        color="text.textDarkCheckbox"
                        cursor="pointer"
                      >
                        Privacy Policy
                      </ChakraLink>
                      , and {''}
                      <ChakraLink
                        href="#"
                        color="text.textDarkCheckbox"
                        cursor="pointer"
                      >
                        Electronics Communication Policy.
                      </ChakraLink>
                    </Text>
                  </Checkbox>
                )}
              />
            </Flex>
          </>
        )}
      </VStack>

      {/* Show API error */}
      <Box mb={7}>
        <Text color="red" textAlign="center" py={2} h={10}>
          {root?.message}
        </Text>
        <Button
          type="submit"
          role="button"
          aria-label={!isRegister ? 'Sign In' : 'Sign Up'}
          colorScheme="primary"
          bg="primary.300"
          textTransform="capitalize"
          form={!isRegister ? 'login-form' : 'register-form'}
          isDisabled={isDisabledSubmitBtn}
          isLoading={isSubmitting}
        >
          {!isRegister ? 'Sign In' : 'Sign Up'}
        </Button>
      </Box>

      <Flex justifyContent="center">
        <Text
          fontWeight="medium"
          textAlign="center"
          dangerouslySetInnerHTML={{
            __html: !isRegister
              ? 'Don&apos;t have an account?'
              : 'Already have an account?',
          }}
        />
        <Button
          as={Link}
          isDisabled={isSubmitting}
          href={!isRegister ? `/${ROUTES.REGISTER}` : `/${ROUTES.LOGIN}`}
          aria-label={!isRegister ? 'sign up' : 'sign in'}
          w="fit-content"
          py={0}
          _hover={{
            bg: 'transparent',
          }}
          color="text.primary"
          bg="transparent"
          fontWeight="semibold"
          textDecoration="underline"
          ml={2}
          onClick={(e) => {
            if (isSubmitting) return e.preventDefault();
          }}
        >
          {!isRegister ? 'Sign Up' : 'Sign In'}
        </Button>
      </Flex>
      <AuthFooter />
    </Box>
  );
};

const AuthForm = memo(AuthFormComponent);

export default AuthForm;
