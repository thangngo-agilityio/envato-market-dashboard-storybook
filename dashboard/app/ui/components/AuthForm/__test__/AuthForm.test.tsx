import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Component
import AuthForm from '@/ui/components/AuthForm';

// Constants
import { ROUTES } from '@/lib/constants';

// Hooks
import { useAuth } from '@/lib/hooks';

// Mocks
import { USER_SIGN_UP, USER_SIGN_IN } from '@/lib/mocks';

const mockSignIn = jest.fn();
const mockSignUp = jest.fn();
const mockRouter = { push: jest.fn() };
const mockSetError = jest.fn();

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  useAuth: jest.fn(),
}));

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  useRouter: () => mockRouter,
}));

describe('AuthForm components', () => {
  beforeAll(() => {
    (useAuth as jest.Mock).mockReturnValue({
      signIn: mockSignIn,
      signUp: mockSignUp,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('match snapshot with login form', () => {
    const { container } = render(<AuthForm />);

    expect(container).toMatchSnapshot();
  });

  it('match snapshot with register form', () => {
    const { container } = render(<AuthForm isRegister />);

    expect(container).toMatchSnapshot();
  });

  it('renders Login form by default', () => {
    render(<AuthForm />);

    expect(screen.getByLabelText('Sign In')).toBeInTheDocument();
  });

  it('renders Register form by default', () => {
    render(<AuthForm isRegister />);

    expect(screen.getByLabelText('Sign Up')).toBeInTheDocument();
  });

  it('successful register', async () => {
    render(<AuthForm isRegister />);

    await userEvent.type(screen.getByPlaceholderText('First name'), 'John');

    await userEvent.type(screen.getByPlaceholderText('Last name'), 'Doe');

    await userEvent.type(
      screen.getByPlaceholderText('Email'),
      'test@example.com',
    );

    await userEvent.type(screen.getByPlaceholderText('Password'), '1@Dzxcvb');

    await userEvent.type(
      screen.getByPlaceholderText('Confirm password'),
      '1@Dzxcvb',
    );

    await userEvent.click(
      screen.getByText(/By creating an account, you're agreeing to our /i),
    );

    await userEvent.click(screen.getByRole('button', { name: 'Sign Up' }));

    await waitFor(() => {
      expect(mockSignUp).toHaveBeenCalledWith(USER_SIGN_UP);

      expect(mockRouter.push).toHaveBeenCalledWith(ROUTES.ROOT);

      expect(mockSetError).not.toHaveBeenCalled();
    });
  });

  it('failed register', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(screen.getByPlaceholderText('First name'), 'John');

      await userEvent.type(screen.getByPlaceholderText('Last name'), 'Doe');

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );

      await userEvent.type(screen.getByPlaceholderText('Password'), '1@Dzxcvb');

      await userEvent.type(
        screen.getByPlaceholderText('Confirm password'),
        'Abcd@1234',
      );

      await userEvent.click(
        screen.getByText(/By creating an account, you're agreeing to our /i),
      );

      await userEvent.click(screen.getByRole('button', { name: 'Sign Up' }));
    } catch (error) {
      await waitFor(() => {
        expect(mockRouter.push).toHaveBeenCalledWith(ROUTES.REGISTER);

        expect(mockSetError).toHaveBeenCalled();
      });
    }
  });

  it('failed login', async () => {
    try {
      render(<AuthForm />);

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );
      await userEvent.type(
        screen.getByPlaceholderText('Password'),
        'Abcd@1234',
      );

      await userEvent.click(screen.getByRole('button', { name: 'Sign In' }));
    } catch (error) {
      await waitFor(() => {
        expect(mockSignIn).toHaveBeenCalledWith(
          {
            ...USER_SIGN_IN,
          },
          false,
        );

        expect(mockRouter.push).toHaveBeenCalledWith(ROUTES.LOGIN);

        expect(screen.getAllByLabelText).toEqual(
          'Email or password is incorrect',
        );

        expect(mockSetError).toHaveBeenCalled();
      });
    }
  });

  it('Wrong format email', async () => {
    try {
      render(<AuthForm />);

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'testexample.com',
      );

      await userEvent.click(screen.getByRole('button', { name: 'Sign In' }));
    } catch (error) {
      await waitFor(() => {
        expect(screen.getAllByLabelText).toEqual('Email is invalid');
      });
    }
  });

  it('Wrong format password', async () => {
    try {
      render(<AuthForm />);

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );
      await userEvent.type(screen.getByPlaceholderText('Password'), '111111');

      await userEvent.click(screen.getByRole('button', { name: 'Sign In' }));
    } catch (error) {
      await waitFor(() => {
        expect(screen.getAllByLabelText).toEqual(
          'Password contains uppercase, lowercase and special characters',
        );
        expect(screen.getAllByLabelText).toEqual(
          'Password must be more than 8 characters',
        );
      });
    }
  });

  it('Haven`t entered email yet', async () => {
    try {
      render(<AuthForm />);

      await userEvent.type(screen.getByPlaceholderText('Email'), '');
      await userEvent.type(screen.getByPlaceholderText('Password'), '111111');
    } catch (error) {
      await waitFor(() => {
        expect(screen.getByRole('button', { name: 'Sign In' })).toHaveAttribute(
          'disabled',
        );
      });
    }
  });

  it('Haven`t entered password yet', async () => {
    try {
      render(<AuthForm />);

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );
      await userEvent.type(screen.getByPlaceholderText('Password'), '');
    } catch (error) {
      await waitFor(() => {
        expect(screen.getByRole('button', { name: 'Sign In' })).toHaveAttribute(
          'disabled',
        );
      });
    }
  });

  it('Haven`t entered confirm password yet', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );
      await userEvent.type(screen.getByPlaceholderText('Password'), '12345');

      await userEvent.type(screen.getByPlaceholderText('Confirm password'), '');
    } catch (error) {
      await waitFor(() => {
        expect(screen.getByRole('button', { name: 'Sign Up' })).toHaveAttribute(
          'disabled',
        );
      });
    }
  });

  it('Haven`t entered First name yet', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(screen.getByPlaceholderText('First name'), '');
    } catch (error) {
      expect(screen.getByRole('button', { name: 'Sign Up' })).toHaveAttribute(
        'disabled',
      );
    }
  });

  it('Haven`t entered Last name yet', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(screen.getByPlaceholderText('Last name'), '');
    } catch (error) {
      expect(screen.getByRole('button', { name: 'Sign Up' })).toHaveAttribute(
        'disabled',
      );
    }
  });

  it('Should render message when Email already exists', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(screen.getByPlaceholderText('First name'), 'John');

      await userEvent.type(screen.getByPlaceholderText('Last name'), 'Doe');

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );

      await userEvent.type(screen.getByPlaceholderText('Password'), '1@Dzxcvb');

      await userEvent.type(
        screen.getByPlaceholderText('Confirm password'),
        '1@Dzxcvb',
      );

      await userEvent.click(
        screen.getByText(/By creating an account, you're agreeing to our /i),
      );

      await userEvent.click(screen.getByRole('button', { name: 'Sign Up' }));
    } catch (error) {
      await waitFor(() => {
        expect(mockRouter.push).toHaveBeenCalledWith(ROUTES.REGISTER);

        expect(screen.getAllByLabelText).toEqual('Email already exists');

        expect(mockSetError).toHaveBeenCalled();
      });
    }
  });

  it('Password does not match', async () => {
    try {
      render(<AuthForm isRegister />);

      await userEvent.type(screen.getByPlaceholderText('First name'), 'John');

      await userEvent.type(screen.getByPlaceholderText('Last name'), 'Doe');

      await userEvent.type(
        screen.getByPlaceholderText('Email'),
        'test@example.com',
      );

      await userEvent.type(screen.getByPlaceholderText('Password'), '1@Dzxcvb');

      await userEvent.type(
        screen.getByPlaceholderText('Confirm password'),
        'Abcd@1234',
      );

      await userEvent.click(
        screen.getByText(/By creating an account, you're agreeing to our /i),
      );

      await userEvent.click(screen.getByRole('button', { name: 'Sign Up' }));
    } catch (error) {
      await waitFor(() => {
        expect(mockRouter.push).toHaveBeenCalledWith(ROUTES.REGISTER);

        expect(screen.getAllByLabelText).toEqual('Password does not match');
      });
    }
  });
});
