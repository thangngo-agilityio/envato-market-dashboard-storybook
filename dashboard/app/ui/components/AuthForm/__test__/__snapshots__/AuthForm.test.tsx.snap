// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm components match snapshot with login form 1`] = `
<div>
  <form
    class="css-f2htex"
    id="login-form"
  >
    <div
      class="chakra-stack css-ylms7g"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-169ufh8"
          data-group="true"
        >
          <input
            class="chakra-input css-10kn41x"
            id="field-:r2:"
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-169ufh8"
          data-group="true"
        >
          <input
            class="chakra-input css-10kn41x"
            id="field-:r3:"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            aria-label="The eye icon"
            class="chakra-input__right-element css-1uzzgp1"
            data-testid="right-icon-input"
            type="button"
          >
            <svg
              class="chakra-icon css-1l546ul"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                />
                <path
                  d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="chakra-stack css-qo2j2d"
      >
        <label
          class="chakra-checkbox css-1elxtiq"
        >
          <input
            aria-disabled="false"
            aria-label="remember"
            class="chakra-checkbox__input"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="checkbox"
            value=""
          />
          <span
            aria-hidden="true"
            class="chakra-checkbox__control css-dnty2r"
          />
          <span
            class="chakra-checkbox__label css-1oeb2oe"
          >
            <p
              class="chakra-text css-10ircq3"
            >
              Remember me
            </p>
          </span>
        </label>
        <a
          aria-label="forgot password"
          class="chakra-button css-1pkuff9"
          href="/forgot-password"
        >
          forgot password?
        </a>
      </div>
    </div>
    <div
      class="css-omcqsq"
    >
      <p
        class="chakra-text css-dz8804"
      />
      <button
        aria-label="Sign In"
        class="chakra-button css-4io6p0"
        disabled=""
        form="login-form"
        role="button"
        type="submit"
      >
        Sign In
      </button>
    </div>
    <div
      class="css-1l4w6pd"
    >
      <p
        class="chakra-text css-25kmip"
      >
        Don't have an account?
      </p>
      <a
        aria-label="sign up"
        class="chakra-button css-nl7ued"
        href="/register"
      >
        Sign Up
      </a>
    </div>
    <div
      class="css-178yklu"
    >
      <p
        class="chakra-text css-88a2rn"
      >
        @ 
        2024
         Bankco. All Rights Reserved.
      </p>
    </div>
  </form>
</div>
`;

exports[`AuthForm components match snapshot with register form 1`] = `
<div>
  <form
    class="css-f2htex"
    id="register-form"
  >
    <div
      class="chakra-stack css-ylms7g"
    >
      <div
        class="chakra-stack css-rluqv0"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <div
            class="chakra-input__group css-169ufh8"
            data-group="true"
          >
            <input
              aria-label="first-name"
              class="chakra-input css-10kn41x"
              id="field-:r6:"
              name="firstName"
              placeholder="First name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <div
            class="chakra-input__group css-169ufh8"
            data-group="true"
          >
            <input
              aria-label="last-name"
              class="chakra-input css-10kn41x"
              id="field-:r7:"
              name="lastName"
              placeholder="Last name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-169ufh8"
          data-group="true"
        >
          <input
            class="chakra-input css-10kn41x"
            id="field-:r8:"
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-169ufh8"
          data-group="true"
        >
          <input
            class="chakra-input css-10kn41x"
            id="field-:r9:"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            aria-label="The eye icon"
            class="chakra-input__right-element css-1uzzgp1"
            data-testid="right-icon-input"
            type="button"
          >
            <svg
              class="chakra-icon css-1l546ul"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                />
                <path
                  d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-169ufh8"
          data-group="true"
        >
          <input
            class="chakra-input css-10kn41x"
            id="field-:ra:"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
            value=""
          />
          <button
            aria-label="The eye icon"
            class="chakra-input__right-element css-1uzzgp1"
            data-testid="right-icon-input"
            type="button"
          >
            <svg
              class="chakra-icon css-1l546ul"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                />
                <path
                  d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-blvggq"
      >
        <label
          class="chakra-checkbox css-1elxtiq"
        >
          <input
            aria-disabled="false"
            class="chakra-checkbox__input"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="checkbox"
            value=""
          />
          <span
            aria-hidden="true"
            class="chakra-checkbox__control css-dnty2r"
          />
          <span
            class="chakra-checkbox__label css-1oeb2oe"
          >
            <p
              class="chakra-text css-1dl3xyd"
            >
              By creating an account, you're agreeing to our 
              <a
                class="chakra-link css-htaipz"
                href="#"
              >
                Privacy Policy
              </a>
              , and 
              <a
                class="chakra-link css-htaipz"
                href="#"
              >
                Electronics Communication Policy.
              </a>
            </p>
          </span>
        </label>
      </div>
    </div>
    <div
      class="css-omcqsq"
    >
      <p
        class="chakra-text css-dz8804"
      />
      <button
        aria-label="Sign Up"
        class="chakra-button css-4io6p0"
        disabled=""
        form="register-form"
        role="button"
        type="submit"
      >
        Sign Up
      </button>
    </div>
    <div
      class="css-1l4w6pd"
    >
      <p
        class="chakra-text css-25kmip"
      >
        Already have an account?
      </p>
      <a
        aria-label="sign in"
        class="chakra-button css-nl7ued"
        href="/login"
      >
        Sign In
      </a>
    </div>
    <div
      class="css-178yklu"
    >
      <p
        class="chakra-text css-88a2rn"
      >
        @ 
        2024
         Bankco. All Rights Reserved.
      </p>
    </div>
  </form>
</div>
`;
